controller/src/main/java/io/pravega/controller/store/stream/StreamMetadataStore.java

```diff
@@ -447,6 +451,41 @@
                                                 final OperationContext context,
                                                 final Executor executor);
 
+    /**
+     * This method is called from Rolling transaction workflow after new transactions that are duplicate of active transactions
+     * have been created successfully in segment store.
+     * This method will update metadata records for epoch to add two new epochs, one for duplicate txn epoch where transactions
+     * are merged and the other for duplicate active epoch.
+     *
+     * @param scope          stream scope
+     * @param name           stream name.
+     * @param sealedTxnEpochSegments sealed segments from intermediate txn epoch with size at the time of sealing
+     * @param txnEpoch       epoch for transactions that need to be rolled over
+     * @param time           timestamp
+     * @param context        operation context
+     * @param executor       callers executor
+     * @return CompletableFuture which upon completion will indicate that we have successfully created new epoch entries.
+     */
+    CompletableFuture<Void> rollingTxnNewSegmentsCreated(final String scope, final String name, Map<Long, Long> sealedTxnEpochSegments,
```

Comment:
Why do we have the same methods for rolling txns in two different interfaces? I'm seeing this method also in `Stream.java`.

Reply:
The controller follows following scheme for metadata store:
There is a metadataStore which exposes methods to be performed on streams. MetadataStore is an uber class that manages all streams and fetches stream specific object and calls the counterpart methods on the specific stream object.